python/openassetio/test/managerValidator/_implementation.py

```diff
@@ -0,0 +1,151 @@
+        """
+        Initializes an instance of this class.
+
+        @param runner `unittest.main` A unittest-compatible test runner.
+
+        @param loader ValidatorTestLoader A unittest-compatible test
+        case loader that injects test fixtures into test case instances.
+        """
+        self.__runner = runner
+        self.__loader = loader
+
+    def executeTests(self, extraArgs, module, fixtures):
+        """
+        Run the tests harness test cases in the provided `module`,
+        passing any `extraArgs` to the test runner.
+
+        @param extraArgs `List[str]` Arguments to pass to the unittest
+        runner.
+
+        @param module `types.ModuleType` Python module containing the
+        test cases to execute.
+
+        @param fixtures `dict` A dictionary containing fixtures for the
+        the tests. See harness.executeSuite for structure.
+
+        @return `bool` `True` if all tests succeeded, `False` otherwise.
+        """
+        # Ensure the loader has the correct fixtures
+        self.__loader.setFixtures(fixtures)
+        # Prepend the "program name" to simulate full argv for unittest.
+        argv = [self._kValidatorHarnessProgramName]
+        if extraArgs:
+            argv.extend(extraArgs)
+        runnerInstance = self.__runner(
+            testLoader=self.__loader, argv=argv, module=module, exit=False)
+        return runnerInstance.result.wasSuccessful()
+
+
+class _ValidatorTestLoader(unittest.loader.TestLoader):
+    """
+    Custom test case loader that injects test harness fixtures into
+    test cases for use in querying and assertions.
+    """
+    def __init__(self, session):
+        """
+        Initializes an instance of this class.
+
+        @param session hostAPI.Session.Session Test harness OpenAssetIO
+        @ref session
+        """
+        self.__session = session
+        self.__fixtures = {}
+        super(_ValidatorTestLoader, self).__init__()
+
+    def loadTestsFromTestCase(self, testCaseClass):
+        """
+        Override of base class to additionally inject fixtures and
+        OpenAssetIO @ref session into test cases.
+
+        Injects the child dict of the `fixtures` dict that corresponds
+        to the test case class and method. If no such dict is available
+        then the injected fixtures for the test case will be `None`.
+
+        @param testCaseClass FixtureAugmentedTestCase Test case class
+        that supports injection of fixtures.
+
+        @return `unittest.suite.TestSuite` The unittest suite to
+        execute.
+        """
+        testCaseNames = self.getTestCaseNames(testCaseClass)
+        loadedSuite = self.suiteClass(
+            testCaseClass(
+                self.__fixtures.get(testCaseClass.__name__, {}).get(testCaseName),
+                self.__session, testCaseName
+            )
+            for testCaseName in testCaseNames)
+        return loadedSuite
+
+    def setFixtures(self, fixtures):
```

Comment:
[question] Curious about (from commit message) "[a]llows harness to be re-used with different modules", is this something we envision?  Or was it mainly to more closely align fixture and suite lifetimes?

Ideally we would have a test for swapping out different fixtures with the same `_ValidatorTestLoader`, but currently that's not possible via the public interface so not a big deal.

Reply:
I was just to try and align the lifetimes. In that, the implementation allowed the harness to be re-run with different modules (which we don't currently do), but those would all have to share fixtures, as there was no way to change the fixtures once the loader was constructed and supplied when the harness was constructed.  Happy to drop this commit though.

Reply:
No it's fine, sounds reasonable to me.